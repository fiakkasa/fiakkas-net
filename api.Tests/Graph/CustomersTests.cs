using HotChocolate.Execution;

namespace api.Tests.Graph;

public class CustomersTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task Customers_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              customers {
                totalCount
                nodes {
                  createdAt
                  href
                  id
                  internalId
                  title
                  updatedAt
                  version
                  portfolioCategories {
                    totalCount
                  }
                  portfolioItems {
                    totalCount
                  }
                  portfolioTechnologyCategories {
                    totalCount
                  }
                }
              }
            }
            """);

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task CustomerById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              customerById(id: "Q3VzdG9tZXIKZzE4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5") {
                createdAt
                href
                id
                internalId
                title
                updatedAt
                version
              }
            }
            """);

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
