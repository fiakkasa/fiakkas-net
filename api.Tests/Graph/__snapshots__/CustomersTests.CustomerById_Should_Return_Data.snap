{
  "data": {
    "customerById": {
      "createdAt": "2024-01-01T00:00:00.000Z",
      "href": "/test",
      "id": "Q3VzdG9tZXIKZzE4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5",
      "internalId": "18e483e4-6961-4b25-88a9-d1d0a5161109",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
